### ISSUE TITLE
QNAP Install

### ISSUE THREAD

**Issue by lmrafael** [brainfoolong/web-ftp-client#7]
Hello.

I recently switched from an **ARM** (without Node.js v4) **Synology** NAS to a **x86 QNAP** NAS. There is no QNAP install support listed, but it is supposed to run in it anyways am I right? The `npm install` executed with no errors or warnings. After running the `./wfc start` command from PuTTY I run into the `Web FTP Client could not start` message.

The **error.log** file has a single entry:

```
 events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE 0.0.0.0:4340
    at Object.exports._errnoException (util.js:907:11)
    at exports._exceptionWithHostPort (util.js:930:20)
    at Server._listen2 (net.js:1250:14)
    at listen (net.js:1286:10)
    at net.js:1395:9
    at nextTickCallbackWith3Args (node.js:453:9)
    at process._tickCallback (node.js:359:17)
    at Function.Module.runMain (module.js:443:11)
    at startup (node.js:139:18)
    at node.js:974:3
```
I am guessing it is not succeeding at acquiring my NAS IP?

Thanks in advance!!

**Comment by brainfoolong** [comment]
This line `listen EADDRINUSE 0.0.0.0:4340` say that the port is already used by another application.
Maybe you have more luck with another port, by using the `config.js` in the root folder.

**Comment by lmrafael** [comment]
I tried setting the port in `config.js` to 4345, 5050, 5099, 9099, and a handful of others. Even tried to set the host to my NAS IP. Every time the `EADDRINUSE` popped up in the `error.log`.

Do you have any other suggestion for me to try on? 🤞

**Comment by brainfoolong** [comment]
Hm. I have no other idea. This error is related to the server binding, it says it cannot listen to an address because it's already in use. Using another host doesn't make any difference here. The main thing that counts is the port, the host is just the allowed host from where you can call the site.

Do you change the port and after that, the error is still the same or does the error show the changed port? One thing i can imagine is that maybe the config don't get loaded properly.

**Comment by lmrafael** [comment]
So... **It is working**, from the very first trial. 🤦‍♂️ 

I ran it and got the message: `Web FTP Client could not start`. Tried to access `IPTOSERVER:4340` anyway. With no success.

**My first mistake:** my NAS is on **https**, my browser auto-completed the address I typed in to include the https, hence the error. I was accessing `https://IPTOSERVER:4340` not `http://IPTOSERVER:4340`. 

But, hey, I was misled by the error message, right? Not quite.

**My second mistake:** I only have an `admin` user, so I got the warning: ` For security reasons, we advise: DO NOT RUN THE SERVER AS ROOT` and the error: `Web FTP Client could not start`. But I never imagined they would be related. To my disgrace, the error was due to the fact that my PID as admin is 0, or so I guess. I am not that well versed in shell stuff. And failed the check in the **./wfc** file.

Yeah, yeah... What about the exception in the **error.log**?

**My final mistake:** I assumed that there was some sort of failure here, Sometimes the `error.log` would come up empty, sometimes with the beforementioned exception. What really happened was: it started fine, I thought that there was an error, I ran it again, got the error I was looking for. Stupid me.

I hope that my mistakes can be of some use.
😅
